src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -141,30 +141,21 @@ private function skimUri(string $uri): string
         if (false === $parsedUrl = parse_url($uri)) {
             throw new InvalidArgumentException(sprintf('The given MongoDB Connection URI "%s" is invalid.', $uri));
         }
-
-        $query = [];
-        if (isset($parsedUrl['query'])) {
-            parse_str($parsedUrl['query'], $query);
-        }
-
-        if (isset($query['collection'])) {
-            $this->options['collection'] = $query['collection'];
-            $queryStringPos = strrpos($uri, $parsedUrl['query']);
-            unset($query['collection']);
-            $prefix = substr($uri, 0, $queryStringPos);
-            $newQuery = http_build_query($query, '', '&', PHP_QUERY_RFC3986);
-            if (empty($newQuery)) {
-                $prefix = rtrim($prefix, '?');
-            }
-            $suffix = substr($uri, $queryStringPos + \strlen($parsedUrl['query']));
-            $uri = $prefix.$newQuery.$suffix;
-        }
-
         $pathDb = ltrim($parsedUrl['path'] ?? '', '/') ?: null;
         if (null !== $pathDb) {
             $this->options['database'] = $pathDb;
         }
 
+        $matches = [];
+        if (preg_match('/^(.*[\?&])collection=([^&#]*)&?(([^#]*).*)$/', $uri, $matches)) {
+            $prefix = $matches[1];
+            $this->options['collection'] = $matches[2];
+            if (empty($matches[4])) {
+                $prefix = substr($prefix, 0, \strlen($prefix) - 1);
+            }
```

Comment:
`strlen` id not needed if you provide a negative length to `substr`

Reply:
oops, updated

CI Passed for `Lock`, failure was on `HttpKernel` (unrelated)